### PR TITLE
practice3 + QUESTION

### PR DESCRIPTION
**Pull request by ddryzhov**
added swagger to all aontrollers, added liquibase for two tables, implmeneted full order logic (two models + dtos + mappers + service + serviceimpl + controller + repos) + question! what about rolerepo? we dont use it anywhere so i dont have the roles saved automatically, you told me that we will use it later but its the last task

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/dto/orderitem/OrderItemDto.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package mate.academy.bookstore.dto.orderitem;
+
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/dto/orderitem/OrderItemDto.java:8]: It's ok to just use @Data on DTOs, but your option is equally fine

File: src/main/java/mate/academy/bookstore/model/Order.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Getter
+@Setter
+@ToString(exclude = "orderItems")
+@EqualsAndHashCode(exclude = "orderItems")
+@SQLDelete(sql = "UPDATE orders SET is_deleted = TRUE WHERE id=?")
+@SQLRestriction("is_deleted=FALSE")
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/model/Order.java]:
```suggestion
@SQLRestriction("is_deleted = FALSE")
```

File: src/main/java/mate/academy/bookstore/service/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    private void clearShoppingCart(ShoppingCart shoppingCart) {
+        shoppingCart.getCartItems().clear();
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public OrderDto updateOrderStatus(Long orderId, Order.Status status) {
+        Order order = orderRepository.findById(orderId)
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/service/OrderServiceImpl.java]: private methods should go after public ones

File: src/main/java/mate/academy/bookstore/service/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                .orElseThrow(() -> new EntityNotFoundException("Order not found"));
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/service/OrderServiceImpl.java]: include relevant details in exception messages to facilitate debugging (like ID here)

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookstore/repository/order/OrderRepository.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package mate.academy.bookstore.repository.order;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.bookstore.model.Order;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstore/repository/order/OrderRepository.java:8]:
```suggestion
```

Comment (MentorMateAcademy) [reply src/main/java/mate/academy/bookstore/repository/order/OrderRepository.java:8]: not mandatory

File: src/main/java/mate/academy/bookstore/repository/order/OrderRepository.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+public interface OrderRepository extends JpaRepository<Order, Long> {
+    Optional<List<Order>> findByUserId(Long userId);
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstore/repository/order/OrderRepository.java]:
```suggestion
    List<Order> findByUserId(Long userId);
```

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookstore/repository/order/OrderRepository.java (hunk 5 above)
Comment (ddryzhov) [reply src/main/java/mate/academy/bookstore/repository/order/OrderRepository.java:8]: well, doesnt that improve code readability?

Comment (ddryzhov) [reply src/main/java/mate/academy/bookstore/repository/order/OrderRepository.java:8]: plus I asked a question about the role repo

**Comment by ddryzhov** [comment]
question! what about rolerepository? we dont use it anywhere so i dont have the roles saved automatically

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**